### DISCUSSION TITLE
Require two-factor auth setting

### DISCUSSION THREAD

**Discussion by bryantsmith24450** [drakkan/sftpgo#940] [Q&A] [answered]
Hello,

Has anyone by chance been able to figure out how to enable the "Require two-factor auth for" setting when creating the user via API?  I've been struggling with this and can't seem to get it to work.  

Thanks in Advance!

**Comment by drakkan** [comment, marked as answer]
Just set `2fa_protocols`, take a look [here](https://sftpgo.stoplight.io/docs/sftpgo/25539182a00f3-add-user) or use the builtin swagger-ui (`http://<your ip:your port>/openapi`)

**Comment by bryantsmith24450** [comment]
I feel like an idiot, i didn't realize it was within filters, so i was trying to create it outside of the filters section which prevented it from working.  See below is the correct way i went about it.

        "filters": {
          "2fa_protocols":[
            "HTTP"
          ]
        }

  Thanks again!

**Reply by drakkan** [reply]
> I feel like an idiot, i didn't realize it was within filters, so i was trying to create it outside of the filters section which prevented it from working. See below is the correct way i went about it.
> 
> ```
>     "filters": {
>       "2fa_protocols":[
>         "HTTP"
>       ]
>     }
> ```
> 
> Thanks again!

No problem, if you can please consider [sponsoring](https://github.com/drakkan/sftpgo#sponsors) the project to ensure it will be sustainable in the long run
